src/propertiesdialog.cpp

```diff
@@ -467,45 +479,136 @@ void PropertiesDialog::setupShortcuts()
+        if (!path.isEmpty() && QFile::exists(path)) {
+            dia.selectFile(path);
+        }
+        else {
+            openAppDir = true;
+        }
+    }
+#ifdef APP_DIR
+    if (openAppDir) {
+        auto appDirStr = QString::fromUtf8(APP_DIR);
+        if (!appDirStr.isEmpty()) {
+            QDir appDir(appDirStr);
+            if (appDir.exists()) {
+                dia.setDirectory(appDir);
+            }
+        }
+    }
+#endif
+
+    if (!dia.exec()) {
         return;
+    }
 
     QString fname = dia.selectedFiles().count() ? dia.selectedFiles().at(0) : QString();
-    if (fname.isNull())
+    if (fname.isNull()) {
         return;
+    }
 
     bookmarksLineEdit->setText(fname);
-    openBookmarksFile(bookmarksLineEdit->text());
+    openBookmarksFile();
 }
 
-void PropertiesDialog::openBookmarksFile(const QString &fname)
+void PropertiesDialog::openBookmarksFile()
 {
+    auto fname = bookmarksLineEdit->text();
+    if (fname.isEmpty()) {
+        return;
+    }
+
     QFile f(fname);
     QString content;
-    if (!f.open(QFile::ReadOnly))
-        content = QString::fromLatin1("<qterminal>\n  <group name=\"group1\">\n    <command name=\"cmd1\" value=\"cd $HOME\"/>\n  </group>\n</qterminal>");
-    else
+    if (!f.open(QFile::ReadOnly)) {
+        content = QString::fromLatin1("<qterminal>\n  <group name=\"Change Directory\">\n    <command name=\"Home\" value=\"cd $HOME\"/>\n  </group>\n  <group name=\"File Manager\">\n    <command name=\"Open here\" value=\"xdg-open $(pwd)\"/>\n  </group>\n</qterminal>\n");
+    }
+    else {
         content = QString::fromUtf8(f.readAll());
+    }
 
     bookmarkPlainEdit->setPlainText(content);
     bookmarkPlainEdit->document()->setModified(false);
 }
 
-void PropertiesDialog::saveBookmarksFile(const QString &fname)
+void PropertiesDialog::saveBookmarksFile()
 {
-    if (!bookmarkPlainEdit->document()->isModified())
+    auto fname = bookmarksLineEdit->text();
+    if (fname.isEmpty()) {
+        return;
+    }
+
+    bool fromAppDir = false;
+#ifdef APP_DIR
+    // if the file is chosen from the app directory, save it to the config directory
+    auto appDirStr = QString::fromUtf8(APP_DIR);
+    if (!appDirStr.isEmpty()) {
+        QFileInfo fInfo(fname);
+        if (fInfo.exists() && fInfo.dir() == QDir(appDirStr)) {
+            QString configDir = Properties::Instance()->configDir();
+            if (!configDir.isEmpty()) {
+                fname = QDir(configDir).absoluteFilePath(fInfo.fileName());
+                fromAppDir = true;
+            }
+        }
+    }
+#endif
+
+    // don't proceed if the bookmarks file exists but isn't from the app directory
+    // and the editor isn't modified
+    if (!fromAppDir
+        && !bookmarkPlainEdit->document()->isModified()
+        && QFile::exists(fname)) {
         return;
+    }
 
     QFile f(fname);
-    if (!f.open(QFile::WriteOnly|QFile::Truncate))
-        qDebug() << "Cannot write to file" << f.fileName();
-    else
```

Comment:
> ... but when saving to qterminal_bookmarks.foo no error is displayed and bookmarks still work, but the file will not show up again to select because of the XML file filter.

The file dialog is intentionally limited to `.xml` (as you'd suggested elsewhere) to encourage the user to add that extension, while manual editing has no limit (because, actually, `.xml` isn't needed) and can be seen as an "advanced usage."

But I guess you already know that because you marked it as resolved.

Reply:
As said, the downside is that such a file cannot be found anymore in the filedialog.

Reply:
I'll add `All Files (*)` to the file dialog. By default, `XML files (*.xml)` will be selected.